### PR TITLE
feat(manager/helmv3): add postUpdateOption to update subchart archives

### LINKED ISSUE #18110: Update helm tarball files, on helm chart repositories with dependencies
### What would you like Renovate to be able to do?

When providing a pull request for a helm chart repository, renovate should also put the tarballs of dependent charts into the pull requests if they aren't ignored via git ignore.

See also this [discussion](https://github.com/renovatebot/renovate/discussions/14751)

here is a little [reproducer](https://github.com/alfsch/helm-tarball-reproducer.git)

### If you have any ideas on how this should be implemented, please tell us here.

On helm dependency updates the following steps should be done:

- Update Chart.yaml with new version
- Execute `helm dependency update`
- if not ignored commit
  - changes in charts/**
  - Chart.yaml
  - Chart.lock

### Is this a feature you are interested in implementing yourself?

No

### PR DESCRIPTION
**Pull request by secustor**
## Changes
Adding a `postUpdateOption` which adds changes done by `helm dependency update` to file changes uploaded by Renovate. 

## Context
Closes #18110

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

https://github.com/secustor/renovate-helm-tarball-reproducer

### CODE REVIEW

File: lib/modules/manager/helmv3/artifacts.spec.ts [hunk 1]
```diff
@@ -211,6 +213,216 @@ describe('modules/manager/helmv3/artifacts', () => {
...
+      await helmv3.updateArtifacts({
+        packageFileName: 'Chart.yaml',
+        updatedDeps,
+        newPackageFileContent: chartFile,
+        config: {
+          postUpdateOptions: ['helmUpdateSubChartArchives'],
+          ...config,
+        },
+      })
+    ).toMatchSnapshot([
```
Comment (viceice) [inline lib/modules/manager/helmv3/artifacts.spec.ts:294]: use toEqual

File: lib/modules/manager/helmv3/artifacts.spec.ts [hunk 2]
```diff
@@ -211,6 +213,216 @@ describe('modules/manager/helmv3/artifacts', () => {
...
+      },
+      {
+        file: {
+          type: 'deletion',
+          path: 'charts/example-1.6.2.tgz',
+        },
+      },
+    ]);
+    expect(execSnapshots).toBeArrayOfSize(2);
+    expect(execSnapshots).toMatchSnapshot();
```
Comment (viceice) [inline lib/modules/manager/helmv3/artifacts.spec.ts:310]: use toMatchObject

File: lib/modules/manager/helmv3/readme.md [hunk 3]
```diff
@@ -64,3 +64,14 @@ For this you use a custom `hostRules` array.
   ],
 }
 ```
+
+### SubChart archives
+
+If updates to sub chart archives are desired, a `postUpdateOption` named `helmUpdateSubChartArchives` can be used.
+After enabling this option archives located in the `/chart` folder are managed by Renovate.
+
+```json5
```
Comment (viceice) [inline lib/modules/manager/helmv3/readme.md:73]: please use normal JSON style

Comment (HonkingGoose) [reply lib/modules/manager/helmv3/readme.md:73]:
```suggestion
```json
```

File: lib/modules/manager/helmv3/readme.md [hunk 4]
```diff
@@ -64,3 +64,14 @@ For this you use a custom `hostRules` array.
   ],
 }
 ```
+
+### SubChart archives
+
+If updates to sub chart archives are desired, a `postUpdateOption` named `helmUpdateSubChartArchives` can be used.
```
Comment (HonkingGoose) [inline lib/modules/manager/helmv3/readme.md:70]:
```suggestion
To get updates for subchart archives put `helmUpdateSubChartArchives` in your `postUpdateOption` configuration.
```

File: lib/modules/manager/helmv3/readme.md [hunk 5]
```diff
@@ -64,3 +64,14 @@ For this you use a custom `hostRules` array.
...
+After enabling this option archives located in the `/chart` folder are managed by Renovate.
```
Comment (HonkingGoose) [inline lib/modules/manager/helmv3/readme.md:71]:
```suggestion
Renovate now updates archives in the `/chart` folder.
```

File: lib/modules/manager/helmv3/readme.md [hunk 6]
```diff
@@ -64,3 +64,14 @@ For this you use a custom `hostRules` array.
   ],
 }
 ```
+
+### SubChart archives
```
Comment (HonkingGoose) [inline lib/modules/manager/helmv3/readme.md:68]:
```suggestion
### Subchart archives
```

The helm docs call it `subchart` or `subcharts` [^helm]. So let's use their terms in our docs. 😉 

[^helm]: https://helm.sh/docs/chart_template_guide/subcharts_and_globals/

**Review by HonkingGoose — CHANGES_REQUESTED** [review]
How about this?

File: lib/modules/manager/helmv3/readme.md [hunk 7]
```diff
@@ -64,3 +64,14 @@ For this you use a custom `hostRules` array.
   ],
 }
 ```
+
+### Subchart archives
+
+To get updates for subchart archives put `helmUpdateSubChartArchives` in your `postUpdateOption` configuration.
```
Comment (HonkingGoose) [inline lib/modules/manager/helmv3/readme.md:70]:
```suggestion
To get updates for subchart archives put `helmUpdateSubChartArchives` in your `postUpdateOptions` configuration.
```

Fixing a typo in the config option name.

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

File: lib/modules/manager/helmv3/artifacts.spec.ts [hunk 8]
```diff
@@ -211,6 +213,251 @@ describe('modules/manager/helmv3/artifacts', () => {
...
+        },
+      },
+      {
+        file: {
+          type: 'deletion',
+          path: 'charts/example-1.6.2.tgz',
+        },
+      },
+    ]);
+    expect(execSnapshots).toBeArrayOfSize(2);
```
Comment (viceice) [inline lib/modules/manager/helmv3/artifacts.spec.ts:264]: we don't need this, next march will do it

File: lib/modules/manager/helmv3/artifacts.spec.ts [hunk 9]
```diff
@@ -211,6 +213,251 @@ describe('modules/manager/helmv3/artifacts', () => {
...
```
Comment (viceice) [inline lib/modules/manager/helmv3/artifacts.spec.ts:316]: same here

File: lib/modules/manager/helmv3/artifacts.spec.ts [hunk 10]
```diff
@@ -211,6 +213,251 @@ describe('modules/manager/helmv3/artifacts', () => {
...
+    ).toEqual([
+      {
+        file: {
+          type: 'addition',
+          path: 'charts/example-1.9.2.tgz',
+          contents: undefined,
+        },
+      },
+    ]);
+    expect(execSnapshots).toBeArrayOfSize(2);
```
Comment (viceice) [inline lib/modules/manager/helmv3/artifacts.spec.ts:361]: same

File: lib/modules/manager/helmv3/artifacts.spec.ts [hunk 11]
```diff
@@ -211,6 +213,251 @@ describe('modules/manager/helmv3/artifacts', () => {
...
+        },
+      },
+      {
+        file: {
+          type: 'deletion',
+          path: 'charts/example-1.6.2.tgz',
+        },
+      },
+    ]);
+    expect(execSnapshots).toBeArrayOfSize(2);
```
Comment (viceice) [inline lib/modules/manager/helmv3/artifacts.spec.ts:413]: same

File: lib/modules/manager/helmv3/artifacts.spec.ts [hunk 12]
```diff
@@ -211,6 +213,251 @@ describe('modules/manager/helmv3/artifacts', () => {
...
+        packageFileName: 'Chart.yaml',
+        updatedDeps,
+        newPackageFileContent: chartFile,
+        config: {
+          postUpdateOptions: ['helmUpdateSubChartArchives'],
+          ...config,
+        },
+      })
+    ).toBeNull();
+    expect(execSnapshots).toBeArrayOfSize(2);
```
Comment (viceice) [inline lib/modules/manager/helmv3/artifacts.spec.ts:450]: same

**Review by HonkingGoose — DISMISSED** [review]
Docs look good to me.

**Review by viceice — APPROVED** [review]

**Pull request merged**